sdk/src/main/java/com/horizen/cryptolibprovider/CommonCircuit.java

```diff
@@ -0,0 +1,40 @@
+package com.horizen.cryptolibprovider;
+
+import com.horizen.provingsystemnative.ProvingSystem;
+import com.horizen.provingsystemnative.ProvingSystemType;
+import com.horizen.utils.BytesUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class CommonCircuit {
+    // Note: supportedSegmentSize should correlate with the snark circuit complexity,
+    // but is always less or equal the one defined in the MC network (maxSegmentSize).
+    private static final int maxSegmentSize = (1 << 18);
+    private static final int supportedSegmentSize = (1 << 18);
```

Comment:
Note: previously the `supportedSegmentSize` was `1 << 17`.
`1 << 18` is needed for CSW circuit, because with previous value CSW PK + VK was out of allowed size.

The drawback for the Cert circuit exists. Previously creation time was 9711ms, verification - 310ms.
Now: creation - 13850ms, verification - 689ms

Reply:
Yes. Can be fixed with https://github.com/HorizenOfficial/zendoo-sc-cryptolib/pull/77#pullrequestreview-843267005